File: app/src/main/java/ca/ivasenko/mobiletest/resources/ui/ResourceListViewModel.java
package ca.ivasenko.mobiletest.resources.ui;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.LiveData;
import android.support.annotation.NonNull;

import java.util.List;

import ca.ivasenko.mobiletest.resources.data.ResourceRepository;
import ca.ivasenko.mobiletest.resources.model.ResourceObject;

public class ResourceListViewModel extends AndroidViewModel
{
    private final ResourceRepository mRepository;
    private LiveData<List<ResourceObject>> mAllResources;

    public ResourceListViewModel(@NonNull Application application)
    {
        super(application);
        mRepository = new ResourceRepository(application);
    }

    public LiveData<List<ResourceObject>> getAllResources(String categoryEid)
    {
        if(mAllResources == null)
            mAllResources = mRepository.getAllResources(categoryEid);
        return mAllResources;
    }
}

File: MobileTest/mobiletestcore/src/main/java/com/quickseries/mobiletestcore/viewModels/ResourceListViewModel.kt
package com.quickseries.mobiletestcore.viewModels

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import com.quickseries.mobiletestcore.data.DemoDatabase
import com.quickseries.mobiletestcore.enums.SortDirection
import java.util.*

class ResourceListViewModel : AndroidViewModel {
    private val database: DemoDatabase

    constructor(application: Application) : super(application) {
        database = DemoDatabase.getInstance(application.applicationContext)
    }

    fun getResources() = database.resourceDao.getResourcesFiltered()

    fun getSortDirection(): SortDirection = database.resourceDao.currentSortDirection

    fun getFiltter(): String = database.resourceDao.currentFilter

    fun toggleSortDirection() {
        if (database.resourceDao.currentSortDirection == SortDirection.ASC) {
            database.resourceDao.sort(SortDirection.DESC)
        } else {
            database.resourceDao.sort(SortDirection.ASC)
        }
    }

    fun filter(text: String) {
        this.database.resourceDao.filter(text)
    }

    fun setCategoryId(categoryId: UUID) {
        this.database.resourceDao.setCategoryId(categoryId)
    }
}